web/src/engine/websites/ElevenToon.ts

```diff
@@ -0,0 +1,62 @@
+import { Tags } from '../Tags';
+import icon from './ElevenToon.webp';
+import { Chapter, DecoratableMangaScraper, type Manga, type MangaPlugin } from '../providers/MangaPlugin';
+import { FetchCSS, FetchRequest, FetchWindowScript } from '../FetchProvider';
+import * as Common from './decorators/Common';
+
+function MangaExtractor(anchor: HTMLAnchorElement) {
+    return {
+        id: anchor.pathname + anchor.search,
+        title: anchor.querySelector('.homelist-title span').textContent.trim()
+    };
+}
+@Common.MangasMultiPageCSS('/bbs/board.php?bo_table=toon_c&type=upd&page={page}', 'ul.homelist li[data-id] a', 1, 1, 0, MangaExtractor)
+@Common.PagesSinglePageJS('img_list', 500)
+@Common.ImageAjax()
+
+export default class extends DecoratableMangaScraper {
+
+    public constructor() {
+        super('11toon', `11toon`, 'http://www.11toon.com', Tags.Language.Korean, Tags.Media.Manga, Tags.Source.Aggregator);
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override async Initialize(): Promise<void> {
+        const uri = new URL(this.URI);
+        const request = new FetchRequest(uri.href);
+        this.URI.href = await FetchWindowScript(request, `window.location.origin`, 1500);
```

Comment:
The `uri` is changed, but not recognised in `ValidateMangaURL` as it still matches against the hardcoded URL

Reply:
Ah, good find. Fixing the Regexp .

Reply:
I am afraid i dont get it.
>  but not recognised in ValidateMangaURL

Isnt ValidateMangaURL supposed to validate, well, manga url? Its not used to validate the website url per se.

Because its working on my end in that case

![image](https://github.com/manga-download/haruneko/assets/10671518/cd09bbd9-4eb0-4b26-b18d-b866da0d1c1e)
![image](https://github.com/manga-download/haruneko/assets/10671518/0fb2dd17-08c9-4a7e-8565-ab1978220204)
![image](https://github.com/manga-download/haruneko/assets/10671518/81de9891-cdd7-430f-a00d-ef50cb7ecaf0)

Ok the regexp can be improved (and i can do that) but it works. I wouldnt have submitted a PR without a working test.

Maybe i should use Common.MangaCSS ? I dont know if this website got a stable url for now. I mainly ported the code.

Reply:
It looked to me like this is a website that changes their domain quite often, so i hinted out that the website URL may deviate with the validation pattern sometime in the future and a manual update of the pattern would still be required by a developer.
With latest commit it is now possible to use `{origin}` and `{hostname}` placeholders in the regex pattern as well.

Reply:
Perfect